packages/integrations/mdx/src/rehype-images-to-component.ts

```diff
@@ -0,0 +1,166 @@
+import type { MarkdownVFile } from '@astrojs/markdown-remark';
+import type { Properties, Root } from 'hast';
+import type { MdxJsxAttribute, MdxjsEsm } from 'mdast-util-mdx';
+import type { MdxJsxFlowElementHast } from 'mdast-util-mdx-jsx';
+import { visit } from 'unist-util-visit';
+import { jsToTreeNode } from './utils.js';
+
+export const ASTRO_IMAGE_ELEMENT = 'astro-image';
+export const ASTRO_IMAGE_IMPORT = '__AstroImage__';
+export const USES_ASTRO_IMAGE_FLAG = '__usesAstroImage';
+
+function createArrayAttribute(name: string, values: (string | number)[]): MdxJsxAttribute {
+	return {
+		type: 'mdxJsxAttribute',
+		name: name,
+		value: {
+			type: 'mdxJsxAttributeValueExpression',
+			value: name,
+			data: {
+				estree: {
+					type: 'Program',
+					body: [
+						{
+							type: 'ExpressionStatement',
+							expression: {
+								type: 'ArrayExpression',
+								elements: values.map((value) => ({
+									type: 'Literal',
+									value: value,
+									raw: String(value),
+								})),
+							},
+						},
+					],
```

Comment:
Is there a reason to generate the mdxast manually? What about generating a string and call `jsToTreeNode` at the end?

I was thinking of some optimizations [on discord](https://discord.com/channels/830184174198718474/845430950191038464/1229787951597097042) where we can skip the mdxast generation altogether and put the string directly in the final transformed code.

Using `jsToTreeNode` today would be a bit slower, but having the string at hand would make migrating to that optimization easier.

Reply:
Ugh nevermind, I mistaken this as JS, but it is editing the JSX instead which makes sense.